src/twisted/web/http.py

```diff
@@ -1236,6 +1239,13 @@ def write(self, data):
 
             for name, values in self.responseHeaders.getAllRawHeaders():
                 for value in values:
+                    if name == b"Content-Length" and isOkay and isCONNECT:
```

Comment:
Is there any case-sensitivity issue here?

Reply:
It occurs to me that filtering these headers might be better-accomplished by doing a pre-processing pass here along the lines of:

1. putting `b"Content-Length"` into a constant of some kind, it is wild that we still use literals everywhere
2. doing this:
```
self.responseHeaders.removeHeader(CONTENT_LENGTH)
```

If we want to still emit a warning rather than silently filter it, we could include a `hasHeader` too.

Reply:
This should be ok.

```python
    def getAllRawHeaders(self) -> Iterator[Tuple[bytes, Sequence[bytes]]]:
        """
        Return an iterator of key, value pairs of all headers contained in this
        object, as L{bytes}.  The keys are capitalized in canonical
        capitalization.
        """
        for k, v in self._rawHeaders.items():
            yield self._canonicalNameCaps(k), v
```

Reply:
Great, thank you for tracking that down. Comment withdrawn :)